tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,298 @@
+package charts
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/pkg/config"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	scanError        = "error"
+	scanPending      = "pending"
+	scanRunning      = "running"
+	scanReporting    = "reporting"
+	CisConfigFileKey = "cis"
+	ClusterScanYaml  = `
+apiVersion: cis.cattle.io/v1
+kind: ClusterScan
+metadata:
+  name: %s #string
+  annotations:
+    {}
+#    key: string
+spec:
+  scanProfileName: %s
+  scheduledScanConfig:
+    retentionCount: 3
+    scanAlertRule:
+#      alertOnComplete: boolean
+#      alertOnFailure: boolean
+#    cronSchedule: string
+  scoreWarning: pass
+  `
+)
+
+type CisConfig struct {
+	ProfileName  string `json:"profileName" yaml:"profileName"`
+	ChartVersion string `json:"chartVersion" yaml:"chartVersion"`
+}
+
+type CisBenchmarkTestSuite struct {
+	suite.Suite
+	cisProfileName      string
+	client              *rancher.Client
+	session             *session.Session
+	project             *management.Project
+	chartInstallOptions *charts.InstallOptions
+}
+
+func (c *CisBenchmarkTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *CisBenchmarkTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	cisConfig := new(CisConfig)
+	config.LoadConfig(CisConfigFileKey, cisConfig)
+	c.cisProfileName = cisConfig.ProfileName
+	require.NotEmptyf(c.T(), c.cisProfileName, "CIS profile name required")
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(c.T(), clusterName, "Cluster name to install is not set")
+
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(c.T(), err)
+
+	if cisConfig.ChartVersion == "" {
+		cisConfig.ChartVersion, err = client.Catalog.GetLatestChartVersion(charts.RancherCisBenchmarkName, catalog.RancherChartRepo)
+		require.NoError(c.T(), err)
+	}
+
+	projectConfig := &management.Project{
```

Comment:
is there a steve object for projects you could use instead? we should not use management / v3 when possible.

Reply:
I referred to this code - https://github.com/rancher/rancher/blob/release/v2.7/tests/v2/validation/charts/gatekeeper_test.go#L57